packages/shared_preferences/shared_preferences_android/lib/shared_preferences_android.dart

```diff
@@ -23,44 +28,54 @@ class SharedPreferencesAndroid extends SharedPreferencesStorePlatform {
 
   @override
   Future<bool> remove(String key) async {
-    return (await _kChannel.invokeMethod<bool>(
-      'remove',
-      <String, dynamic>{'key': key},
-    ))!;
+    return _api.remove(key);
   }
 
   @override
   Future<bool> setValue(String valueType, String key, Object value) async {
-    return (await _kChannel.invokeMethod<bool>(
-      'set$valueType',
-      <String, dynamic>{'key': key, 'value': value},
-    ))!;
+    return await _handleSetValue(valueType, key, value) ?? false;
   }
 
   @override
-  Future<bool> clear() async {
+  Future<bool> clear() {
     return clearWithPrefix(_defaultPrefix);
   }
 
   @override
   Future<bool> clearWithPrefix(String prefix) async {
-    return (await _kChannel.invokeMethod<bool>(
-      'clearWithPrefix',
-      <String, dynamic>{'prefix': prefix},
-    ))!;
+    return _api.clearWithPrefix(prefix);
   }
 
   @override
-  Future<Map<String, Object>> getAll() async {
+  Future<Map<String, Object>> getAll() {
     return getAllWithPrefix(_defaultPrefix);
   }
 
   @override
   Future<Map<String, Object>> getAllWithPrefix(String prefix) async {
-    return (await _kChannel.invokeMapMethod<String, Object>(
-          'getAllWithPrefix',
-          <String, dynamic>{'prefix': prefix},
-        )) ??
-        <String, Object>{};
+    final Map<String?, Object?> data = await _api.getAllWithPrefix(prefix);
+    return data.cast<String, Object>();
+  }
+
+  // Call the function according to the type of value provided
+  Future<bool?> _handleSetValue(
```

Comment:
I missed this before; it can return a non-nullable bool, and the `?? false` can be removed from the caller.

Reply:
Actually, this whole method should be removed and inlined into `setValue`; it's only called once, and there's no other code in the caller, so the indirection to a helper method isn't serving any purpose. (I think this was a comment I made in the original PR in the plugins repo that got lost, and didn't notice again until now.)